src/language-helper.ts

```diff
@@ -92,10 +92,26 @@ export class AppLocalization {
         this.resetLocalizedUI();
     }
 
+    private denormalize(locale: string): string {
+        if (locale === "en") {
+            locale = "en-EN";
+        }
+        if (locale.indexOf("-") >= 0) {
+            const langDesc = 1;
+            const partsReq = 2;
+            const part = locale.split("-");
+            if (part.length >= partsReq) {
+                part[langDesc] = part[langDesc].toUpperCase();
+            }
+            return part.join("_");
+        }
+        return locale;
```

Comment:
A few things about this function:
 * The `if` branch is redundant? The code only does the split if the string contains a hyphen, so we know we'll get at least two parts.
 * Having `langDesc` & `partsReq` as defined constants I think might be adding more confusion. I'd probably just use the numbers straight in the code (if `partsReq` is even necessary).
 * Maybe `parts` instead of `part` for the variable - it's all the parts of the split string
 * Could it get a comment saying what it does?

Reply:
- The first if was like a guard to avoid variables to be reserved if there is no hyphen, maybe those will be optimized out?
- maybe langDesc rename to localeIndex. Currently the second part is for uppercase. 
If there are cases like aa_bb_cc, then it would output aa-BB-cc, using last index: aa-bb-CC.
- Parts would be a better name
- partsReq can be removed, if change the > langDesc is changed to comparison

Reply:
Added comment 
Removed the 'if' check
Used fixed index numbers instead of defined constant